plugins/configurationcache/openravepy_configurationcache.cpp

```diff
@@ -193,6 +193,8 @@ typedef OPENRAVE_SHARED_PTR<PyConfigurationCache> PyConfigurationCachePtr;
 OPENRAVE_PYTHON_MODULE(openravepy_configurationcache)
 {
     using namespace configurationcachepy;
+#undef NUMPY_IMPORT_ARRAY_RETVAL
+#define NUMPY_IMPORT_ARRAY_RETVAL
```

Comment:
@cielavenir i don't understand this, can you comment on it?

Reply:
That NUMPY_IMPORT_ARRAY_RETVAL comes from https://github.com/numpy/numpy/commit/2fe642a524a4bd2f870e7def51140f94d136af73 , which exists until 1.18.x.

This import_array is meant to be called in `init##modname` (Python2) or `PyInit_##modname` (Python3). As these functions return types are different, import_array macro implementation is different in Python2 version and Python3 versions.

This concept is incompatible with [PYBIND11_MODULE](https://github.com/pybind/pybind11/blob/master/include/pybind11/detail/common.h#L332) (and [BOOST_PYTHON_MODULE](https://github.com/boostorg/python/blob/master/include/boost/python/module_init.hpp)).
There are several works to get rid of this matter such as ethz-asl/numpy_eigen#6 and https://github.com/EttusResearch/uhd/blob/master/host/python/pyuhd.cpp#L53 .

Another idea is to import oldnumeric.h https://github.com/numpy/numpy/blob/main/numpy/core/include/numpy/oldnumeric.h

Note that I locally tested numpy 1.18.5. I'm not sure if numpy 1.19+ are supported or not.

Reply:
It seems that Gary manually expanded the macro.
https://github.com/tgn3000/openrave3/blob/python3/src/bindings/openravepy_int.cpp#L2254

Reply:
Well it would be possible to cherry-pick https://github.com/tgn3000/openrave3/commit/2fcaccab669a6da9481913bfce75b8b06afb7309